Parser/Parser.cs

```diff
@@ -61,6 +62,17 @@ NBlock Block ()
       return vars.ToArray ();
    }
 
+    NConstDecl[] ConstDecl () {
+      List<NConstDecl> vars = new ();
+      while (!Peek (VAR)) {
+         var name = Expect (IDENT);
+         Expect (EQ);
+         var val = mToken;
+         if (Match (mToken.Kind) && Match (SEMI)) { vars.Add (new NConstDecl (name, new NLiteral(val))); } 
+         else { mToken = Prev; Expect (SEMI); }

```

Comment:
You are back tracking here. If you set mToken to Prev then you have to set Prev to PrevPrev. But you cannot set PrevPrev as that value is lost. You should not do that.